enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,442 @@
+>Guidance in the early 4.x time frame has been around the cluster admin using the MCO post install to effectively update the
+>filesystem of each node with the necessary subscription related files in a well known location that CRI-O would look
+>for.  There is concern though using this through MCO from a couple of perspectives.  One, the MCO api is not
+>natural for someone who wants to do something higher level like "enable entitlements for builds".  Second, using the
+>MCO for any node update requires a restart of the node, which is onerous for this scenario.  Also note, the RHCOS
+>team want to disable their current default setting of auto-mounting secrets from the host (i.e. stuff from 
+>`/etc/share/containers/mounts.conf`), since RHCOS does not fully install subscription manager (and never will), and they have 
+>problems when only partial subscription manager metadata is available.  [RHCOS has the subscription-manager-certs 
+>installed but are missing the subscription-manger package which carries the rhsm.conf.](https://bugzilla.redhat.com/show_bug.cgi?id=1783393)
+>They are entertaining an MCO solution to work around this problem, but disabling auto-mount, and then having a 
+>solution which adds all the subscription manager in one fell swoop, would be their preference.  In the end, providing
+>an alternative to an MCO based solution **would seem** to facilitate RHCOS's future intentions and it would seem we should
+>minimally coordinate with them to facilitate how they stage future changes  
+> 2. There has been discussion elsewhere around having a more generic notion of "global resources" that any pod can
+>mount.  There is an open question of whether this proposal should tackle that.  Would we gate this enhancement on the 
+>availability of such a feature (probably "not sure")?  Upstream feature work could be implied with such an item.  But
+>what influence such a notion could have on this proposal is TBD. 
+> 3. After 4.5 epic planning it was made clear that a more general solution around entitlements, beyond just builds,
+>is desired.  PM has the to-do to investigate, prioritize, and create/assign epics accordingly.  Investigating in the 
+>interim around build specific needs via this proposal has been deemed acceptable, but certainly a subsequent, broader
+>proposal may necessitate adjustments to this one.  This proposal at the moment is straddling the fence between speaking
+>to a generic solution vs. a build specific one (though things are trending toward the former).  But a rebranding of 
+>this proposal may occur during the review process.
+> 4. We did *NOT* provide for multiple sets of entitlement credentials in 3.x.  There have been no official, RFE style  
+>registered requirements entitlements at a namespace level.  However, we've started to get (slightly) less official
+>feedback that customers want this in a more automated sense (vs. the current manual steps).  At a minimum, and is
```

Comment:
This should be as simple as possible, I have customers who just started with OpenShift, they can work with `oc` and some sample `yaml` files but are completely overwhelmed with the entitlement story. Especially if they are used only to satellite and expect things just to work when they subscribe.